stl/src/atomic.cpp

```diff
@@ -33,4 +39,281 @@ _CRTIMP2_PURE void __cdecl _Unlock_shared_ptr_spin_lock() { // release previousl
 #endif // _M_ARM
 }
 
+static std::once_flag _Atomic_spin_count_initialized;
+
+static long _Atomic_spin_count;
+
+static void _Atomic_spin_count_initialize() {
+    _Atomic_spin_count = (std::thread::hardware_concurrency() == 1 ? 0 : 10'000);
+}
+
+enum _Atomic_spin_phase {
+    _ATOMIC_SPIN_PHASE_MASK              = 0xF000'0000,
+    _ATOMIC_WAIT_PHASE_MASK              = 0x0F00'0000,
+    _ATOMIC_SPIN_VALUE_MASK              = 0x00FF'FFFF,
+    _ATOMIC_WAIT_PHASE_SPIN              = 0x0000'0000,
+    _ATOMIC_WAIT_PHASE_WAIT_SET          = 0x0100'0000,
+    _ATOMIC_WAIT_PHASE_WAIT_CLEAR        = 0x0200'0000,
+    _ATOMIC_WAIT_PHASE_WAIT_NO_SEMAPHORE = 0x0300'0000,
+    _ATOMIC_SPIN_PHASE_INIT_SPIN_COUNT   = 0x0000'0000,
+    _ATOMIC_SPIN_PHASE_INIT_SPIN         = 0x1000'0000,
+    _ATOMIC_SPIN_PHASE_INIT_SWITCH_THD   = 0x2000'0000,
+    _ATOMIC_SPIN_PHASE_INIT_SLEEP_ZERO   = 0x3000'0000,
+    _ATOMIC_SPIN_PHASE_INIT_SLEEP        = 0x4000'0000,
+
+    _ATOMIC_SPIN_MASK = _ATOMIC_SPIN_PHASE_MASK | _ATOMIC_SPIN_VALUE_MASK,
+};
+
+bool __cdecl _Atomic_spin_active_only(long& _Spin_context) {

```

Comment:
Do you have a suitable "satellite" DLL already to add these exports to?
Also inlining of those functions probably could be considered.

Reply:
No, once we ship one we can't change its export surface ever again; it's the same problem. As soon as anyone could have app-locally deployed one of our DLLs we have to assume that the one in the process might be the oldest one ever shipped which wouldn't have the new exports you want.

The one that we use to export `default_memory_resource` is probably the simplest one to use as a model since it only has a couple of exports. https://github.com/microsoft/STL/blob/08ed0de1a688935cfe6057b5f6cb86e10dbb9a04/stl/CMakeLists.txt#L430 is the associated build system stuff for it.

Reply:
Thanks, will put it in a new DLL.
Probably will need to try other C++20 concurrency stuff to avoid adding unnecessary amount of DLLs.